fs/dafsserver/dafsserver.cpp

```diff
@@ -1396,21 +1460,413 @@ class CRemoteDiskReadActivity : public CRemoteDiskBaseActivity
     {
         return prefetchBuffer.tell() + baseFpos;
     }
-    virtual unsigned __int64 getLocalFilePosition(const void * row) override
+};
+
+
+class CRemoteCsvReadActivity : public CRemoteStreamReadBaseActivity
+{
+    typedef CRemoteStreamReadBaseActivity PARENT;
+
+    StringBuffer csvQuote, csvSeparate, csvTerminate, csvEscape;
+    unsigned __int64 headerLines = 0;
+    unsigned __int64 maxRowSize = 0;
+    bool preserveWhitespace = false;
+    CSVSplitter csvSplitter;
+    Owned<const IDynamicFieldValueFetcher> fieldFetcher;
```

Comment:
Cleaner as Owned<CFieldFetcher> - would avoid a cast elsewhere.

Reply:
well the cast (there's 1) is casting to non-const, but it's also/mainly casting to the impl. the interface doesn't have a setCurrentMatch  (which is specific to the xml impl).